core/templates/local_vector.h

```diff
@@ -137,10 +135,16 @@ class LocalVector {
 	}
 	_FORCE_INLINE_ bool is_empty() const { return count == 0; }
 	_FORCE_INLINE_ U get_capacity() const { return capacity; }
-	_FORCE_INLINE_ void reserve(U p_size) {
-		p_size = tight ? p_size : nearest_power_of_2_templated(p_size);
+	void reserve(U p_size) {
 		if (p_size > capacity) {
-			capacity = p_size;
+			if (tight) {
```

Comment:
Please abstract this calculation into an inlined function instead of copying the code around.